staging/src/k8s.io/apiserver/pkg/server/openapi/openapi.go

```diff
@@ -17,35 +17,63 @@ limitations under the License.
 package openapi
 
 import (
+	"bytes"
+	"compress/gzip"
+	"crypto/sha512"
+	"encoding/json"
 	"fmt"
+	"gopkg.in/yaml.v2"
 	"net/http"
 	"reflect"
 	"strings"
 
 	"github.com/emicklei/go-restful"
 	"github.com/go-openapi/spec"
+	"github.com/golang/protobuf/proto"
+	"github.com/googleapis/gnostic/OpenAPIv2"
+	"github.com/googleapis/gnostic/compiler"
 
 	"k8s.io/apimachinery/pkg/openapi"
-	"k8s.io/apimachinery/pkg/util/json"
 	genericmux "k8s.io/apiserver/pkg/server/mux"
 	"k8s.io/apiserver/pkg/util/trie"
 )
 
 const (
 	OpenAPIVersion  = "2.0"
 	extensionPrefix = "x-kubernetes-"
+
+	ContentType = "Content-Type"
+	MIME_JSON   = "application/json"
+	MIME_PB     = "application/com.github.googleapis.gnostic.OpenAPIv2+protobuf"
+	MIME_PB_GZ  = "application/x-gzip"
+	MIME_BINARY = "application/octet-stream"
```

Comment:
Unfortunately (or fortunately) we don't need this anymore after etag implementation.